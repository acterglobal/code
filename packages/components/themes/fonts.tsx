import '@fontsource/montserrat/300.css'
import '@fontsource/montserrat/400.css'
import '@fontsource/montserrat/500.css'
import '@fontsource/montserrat/600.css'
import '@fontsource/montserrat/700.css'
import '@fontsource/montserrat/800.css'
import '@fontsource/montserrat/900.css'
import '@fontsource/montserrat/index.css'

export const typography = {
  fontFamily: 'Montserrat',
  fontWeightLight: 400,
  fontWeightRegular: 500,
  fontWeightMedium: 600,
  fontWeightBold: 700,
}
