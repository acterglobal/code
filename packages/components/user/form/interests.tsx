import React, { FC } from 'react'
import { Form as FormikForm, Formik } from 'formik'
import {
  InterestsAddSection,
  InterestAddSectionValues,
} from '@acter/components/acter/form/interests-add-section'
import { ProfileFormLayout } from '@acter/components/user/form/layout'
import { FormButtons } from '@acter/components/util/forms'
import { InterestType } from '@acter/schema'
import { useUser } from '@acter/lib/user/use-user'
import { useUpdateActer } from '@acter/lib/acter/use-update-acter'

export interface ProfileInterestsFormProps {
  interestTypes: InterestType[]
}

export const ProfileInterestsForm: FC<ProfileInterestsFormProps> = ({
  interestTypes,
}) => {
  const { user, loading } = useUser()
  const [updateActer] = useUpdateActer(user?.Acter)
  if (loading || !user) {
    return <>Loading...</>
  }

  const handleSubmit = (values, _actions) => updateActer(values)
  const initialValues: InterestAddSectionValues = {
    interestIds:
      user.Acter?.ActerInterests?.map(({ Interest }) => Interest.id) || [],
  }
  return (
    <ProfileFormLayout>
      <Formik initialValues={initialValues} onSubmit={handleSubmit}>
        <FormikForm>
          <InterestsAddSection interestTypes={interestTypes} />
          <FormButtons align="right" hideUnlessDirty={true} />
        </FormikForm>
      </Formik>
    </ProfileFormLayout>
  )
}
