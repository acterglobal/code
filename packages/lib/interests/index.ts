export * from '@acter/lib/interests/get-selected-interests'
export * from '@acter/lib/interests/get-toplevel-types'
export * from '@acter/lib/interests/get-selected-toplevel-interest-types'
