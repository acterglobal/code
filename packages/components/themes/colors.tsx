import { CSSProperties } from 'react'

import { ActivityTypes, InterestTypes } from '@acter/lib/constants'

export type Color = CSSProperties['color']

type Grey = {
  extraLight: Color
  light: Color
  main: Color
  dark: Color
}

type Blue = {
  dark: Color
  light: Color
}

type ActivityTypeColor = {
  [key in ActivityTypes]: Color
}

type ActivityTypeDarkColor = {
  dark: ActivityTypeColor
}

type ActivityTypeColors = ActivityTypeColor & ActivityTypeDarkColor

type InterestTypeColors = {
  [key in InterestTypes]: Color
}

type OtherColors = {
  searchTab: Color
  notificationBadge: Color
}

type ContentColours = {
  title: Color
  description?: Color
}

export type Colors = {
  black: Color
  blue: Blue
  grey: Grey
  white: Color
  red: Color
  error: Color
  activityTypes: ActivityTypeColors
  interestTypes: InterestTypeColors
  others: OtherColors
  content: ContentColours
}

export const paletteColors = {
  primary: {
    main: '#1EB001',
  },
  secondary: {
    main: '#1B2941',
    dark: '#071528',
    light: '#D5D5D5',
    contrastText: '#D5D5D5',
  },
  background: {
    default: '#ECEFF4',
  },
}

export const colors: Colors = {
  black: '#000',
  white: '#fff',
  red: '#FF6754',
  error: '#EA4335',
  blue: {
    dark: '#243141',
    light: '#3A527D',
  },
  grey: {
    extraLight: '#E3E8F2',
    light: '#D9D9D9',
    main: '#929292',
    dark: '#5E5E5E',
  },
  activityTypes: {
    event: '#2A2A2A',
    project: '#67A24A',
    idea: '#ECAC3C',
    meeting: '#5778B2',
    dark: {
      event: '#4287f5',
      project: '#25cc04',
      idea: '#ffab03',
      meeting: '#D5D5D5',
    },
  },
  interestTypes: {
    Economy: '#FFC53B',
    Environment: '#69B656',
    Social: '#F29C40',
    Approach: '#545559',
    Focus: '#000',
    Tags: '#000',
  },
  others: {
    searchTab: 'rgba(196, 196, 196, 0.18)',
    notificationBadge: '#E15877',
  },
  content: {
    title: '#171717',
  },
}
