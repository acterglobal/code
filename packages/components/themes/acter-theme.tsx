import React, { FC } from 'react'

import { ThemeProvider } from '@material-ui/core/styles'
import { createTheme, responsiveFontSizes } from '@material-ui/core/styles'

import { colors, Colors, paletteColors } from '@acter/components/themes/colors'
import { typography } from '@acter/components/themes/fonts'

declare module '@material-ui/core/styles/createTheme' {
  interface Theme {
    colors: Colors
  }
  interface ThemeOptions {
    colors: Colors
  }
}

declare module '@material-ui/core/styles/createMixins' {
  interface SidebarMixin {
    primaryWidth: number
    secondaryWidth: number
  }
  interface Mixins {
    sidebar: SidebarMixin
  }
}

export const theme = createTheme({
  palette: paletteColors,

  colors: colors,

  typography: typography,

  props: {
    MuiButton: {
      disableElevation: true,
      disableRipple: true,
    },
  },

  mixins: {
    sidebar: {
      primaryWidth: 8,
      secondaryWidth: 22,
    },
  },
})

export const acterTheme = responsiveFontSizes(theme)

export const ActerThemeProvider: FC = ({ children }) => (
  <ThemeProvider theme={acterTheme}>{children}</ThemeProvider>
)
