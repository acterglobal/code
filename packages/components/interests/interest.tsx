import React, { FC } from 'react'
import { Interest as InterestType } from '@schema'
import { Box, Typography } from '@material-ui/core'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import Image from 'next/image'
import clsx from 'clsx'
import { disabledColor, interestColors } from '@acter/components/themes/colors'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    chip: {
      //   margin: 3,
      margin: theme.spacing(0.2),
      width: theme.spacing(23),
      height: theme.spacing(3),
      border: ({ type }: { type: string }) =>
        type === 'Tags' ? 'none' : '1px solid',
      borderRadius: theme.spacing(3),
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      cursor: 'pointer',
    },
    // TODO: refactor to use InterestType-specific classes
    outline: {
      borderColor: ({ type }: { type: string }) => interestColors[type],
      color: ({ type }: { type: string }) => interestColors[type],
    },
    selected: {
      border: ({ type }: { type: string }) =>
        type === 'Tags' ? '1px solid' : 'none',

      borderColor: ({ type }: { type: string }) =>
        type === 'Tags' && interestColors[type],

      backgroundColor: ({ type }: { type: string }) =>
        type !== 'Tags' ? interestColors[type] : '',

      color: ({ type }: { type: string }) =>
        type === 'Tags' ? interestColors[type] : 'white',
    },
    disable: {
      color: disabledColor,
      borderColor: disabledColor,
      cursor: 'default',
    },
    name: {
      marginLeft: theme.spacing(1),
      fontSize: '0.7rem',
    },
    rightSideBox: {
      display: 'flex',
      alignItems: 'center',
    },
    number: {
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
    },
  })
})

export interface InterestProps {
  interest: InterestType
  type: string
  selected?: boolean
  disabled?: boolean
  onSelectedInterestsChange?: (interest: string, type: string) => void
}

export const Interest: FC<InterestProps> = ({
  interest,
  type,
  selected = true,
  disabled = false,
  onSelectedInterestsChange,
}) => {
  const classes = useStyles({ type })

  if (!interest) {
    return null
  }

  const handleClick = () => {
    if (disabled && !selected) return null
    else onSelectedInterestsChange(interest.id, type)
  }

  return (
    <Box
      className={clsx(
        classes.chip,
        selected
          ? classes.selected
          : disabled
          ? classes.disable
          : classes.outline
      )}
      onClick={handleClick}
      role="listitem"
    >
      <Typography className={classes.name} variant="caption">
        {type === 'Tags' ? `# ${interest.name}` : interest.name}
      </Typography>

      {interest.sdgNumber && (
        <Box className={clsx(classes.rightSideBox)}>
          <Image src="assets/SDG-logo.png" alt="SDG" width={15} height={15} />
          <Typography className={classes.number} variant="caption">
            {interest.sdgNumber}
          </Typography>
        </Box>
      )}
    </Box>
  )
}
