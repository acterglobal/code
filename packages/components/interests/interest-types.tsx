import React, { FC } from 'react'
import {
  Box,
  Divider,
  Typography,
  makeStyles,
  createStyles,
  Theme,
} from '@material-ui/core'
import { InterestType } from '@acter/schema/types'
import { Interest } from '@acter/components/interests/interest'
import { Size, InterestTypes as InterestTypeName } from '@acter/lib/constants'

const { FOCUS, ECONOMY, ENVIRONMENT, SOCIAL } = InterestTypeName
export interface InterestTypesProps {
  type: InterestType
  allTypes: InterestType[]
  selectedInterests?: string[]
  selectedTypes?: InterestTypeName[]
  disabled?: boolean
  showTitle?: boolean
  showSubTypeTitles?: boolean
  columns?: boolean
  divider?: boolean
  chipSize?: Size
  onSelectedInterestsChange?: (interest: string, type: string) => void
}

export const InterestTypes: FC<InterestTypesProps> = ({
  type,
  allTypes,
  onSelectedInterestsChange,
  selectedInterests,
  selectedTypes,
  disabled,
  showTitle = false,
  showSubTypeTitles = true,
  columns = false,
  divider: divider = false,
  chipSize,
}) => {
  const typeName: string = type.name
  const classes = useStyles({ typeName })

  const subTypes = allTypes.filter(
    (subtype) => type.id === subtype.parentInterestTypeId
  )

  if (subTypes.length > 0) {
    return (
      <>
        {subTypes.map((subType) => (
          <InterestTypes
            key={subType.id}
            type={subType}
            allTypes={subTypes}
            onSelectedInterestsChange={onSelectedInterestsChange}
            selectedInterests={selectedInterests}
            selectedTypes={selectedTypes}
            showTitle={true && showSubTypeTitles}
            disabled={
              selectedTypes &&
              selectedTypes.filter((selectedType) =>
                [FOCUS, ENVIRONMENT, SOCIAL, ECONOMY].includes(selectedType)
              ).length >= 5
            }
            chipSize={chipSize}
            columns={columns}
          />
        ))}
        {type.name === FOCUS && (
          <>
            {divider && (
              <Divider variant="middle" className={classes.divider} />
            )}
            <Box ml={3}>
              <Interest
                interest={type.Interests[0]}
                type={type.name}
                chipSize={chipSize}
                onSelectedInterestsChange={onSelectedInterestsChange}
                selected={
                  selectedInterests &&
                  selectedInterests.includes(type.Interests[0].id)
                }
                disabled={
                  disabled ||
                  (selectedTypes &&
                    selectedTypes.filter((selectedType) =>
                      [FOCUS, ENVIRONMENT, SOCIAL, ECONOMY].includes(
                        selectedType
                      )
                    ).length >= 5)
                }
              />
            </Box>
          </>
        )}
      </>
    )
  } else {
    return (
      <Box ml={columns ? 0 : 3} role="list">
        {showTitle && (
          <Typography className={classes.title}>{type.name}</Typography>
        )}
        <Box
          style={
            columns
              ? { display: 'flex', flexDirection: 'column' }
              : { display: 'flex', flexWrap: 'wrap' }
          }
        >
          {type.Interests.map((interest) => {
            return (
              <Box
                key={interest.id}
                style={{
                  position: 'relative',
                  display: columns ? 'block' : 'inline-block',
                }}
              >
                <Interest
                  interest={interest}
                  type={type.name as InterestTypeName}
                  chipSize={chipSize}
                  onSelectedInterestsChange={onSelectedInterestsChange}
                  selected={
                    selectedInterests && selectedInterests.includes(interest.id)
                  }
                  disabled={
                    disabled ||
                    (selectedTypes &&
                      selectedTypes.filter(
                        (selectedType) => selectedType === type.name
                      ).length >= 5)
                  }
                />
              </Box>
            )
          })}
        </Box>
      </Box>
    )
  }
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    divider: {
      marginLeft: theme.spacing(3),
      marginRight: theme.spacing(0.8),
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(1),
      backgroundColor: theme.colors.grey.main,
    },
    title: {
      color: ({ typeName }: { typeName: string }) =>
        theme.colors.interestTypes[typeName],
    },
  })
)
