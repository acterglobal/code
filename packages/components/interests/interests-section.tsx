import React, { FC } from 'react'

import { Box, Typography } from '@material-ui/core'

import { LoadingSpinner } from '@acter/components/atoms/loading/spinner'
import { InterestTypes } from '@acter/components/interests/interest-types'
import { Size } from '@acter/lib/constants'
import {
  getSelectedTopLevelTypes,
  getSelectedInterests,
} from '@acter/lib/interests'
import { useInterestTypes } from '@acter/lib/interests/use-interest-types'
import { capitalize } from '@acter/lib/string/capitalize'
import { Interest } from '@acter/schema'

export interface InterestsSectionProps {
  selected?: Interest[]
}

export const InterestsSection: FC<InterestsSectionProps> = ({ selected }) => {
  const { interestTypes, fetching: interestTypesLoading } = useInterestTypes()

  if (interestTypesLoading) return <LoadingSpinner />
  if (!interestTypes) return null

  const typesWithSelectedInterests = getSelectedInterests(
    interestTypes,
    selected
  )
  const selectedTopLevel = getSelectedTopLevelTypes(
    typesWithSelectedInterests,
    selected
  )

  return (
    <Box style={{ display: 'flex' }}>
      {selectedTopLevel.map((type) => (
        <Box key={type.id} role="list">
          <Typography
            id="interest-type-name"
            style={{ marginLeft: 4, fontWeight: 600 }}
          >
            {capitalize(type.name)}
          </Typography>
          <InterestTypes
            type={type}
            chipSize={Size.SMALL}
            allTypes={typesWithSelectedInterests}
            onSelectedInterestsChange={() => null}
            columns={true}
            showSubTypeTitles={false}
          />
        </Box>
      ))}
    </Box>
  )
}
