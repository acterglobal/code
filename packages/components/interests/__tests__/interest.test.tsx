import React from 'react'
import { render } from '@acter/lib/test-utils'
import { Interest } from '@acter/components/interests/interest'
import { ExampleInterest } from '@acter/schema/fixtures/interest/example-interests'
import { ActerThemeProvider } from '@acter/components/themes/acter-theme'

describe('[Interest]', () => {
  it('should render Tag interest with correct info and styles', () => {
    const { container } = render(
      <ActerThemeProvider>
        <Interest
          interest={ExampleInterest}
          type={ExampleInterest.InterestType.name}
        />
      </ActerThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
