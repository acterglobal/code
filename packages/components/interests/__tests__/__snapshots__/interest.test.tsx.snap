// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Interest] should render Tag interest with correct info and styles 1`] = `
<div>
  <div
    class="MuiBox-root-11 MuiBox-root-12 makeStyles-chip-1 makeStyles-chip-8 makeStyles-selected-3 makeStyles-selected-10"
    role="listitem"
  >
    <span
      class="MuiTypography-root-13 makeStyles-name-5 MuiTypography-caption-16"
    >
      Example Interest
    </span>
  </div>
</div>
`;
